presto-main/src/main/java/io/prestosql/execution/QueryStateMachine.java

```diff
@@ -761,6 +762,14 @@ public boolean transitionToFinishing()
             return false;
         }
 
+        try {
+            cleanupQuery();
+        }
+        catch (Exception e) {
+            transitionToFailed(e);
```

Comment:
Please extract changes in QueryStateMachine.java as separate commit: "Handle exceptions from Metadata.cleanupQuery"

cc @dain

Reply:
I actually suggested he could squash it, but you're right, it is better as a separate commit